operator/pkg/model/translation/translator_test.go

```diff
@@ -188,6 +188,33 @@ func TestSharedIngressTranslator_getServices(t *testing.T) {
 	}
 }
 
+func TestSharedIngressTranslator_getHTTPRouteListenerProxy(t *testing.T) {
+	i := &defaultTranslator{
+		name:             "cilium-ingress",
+		namespace:        "kube-system",
+		secretsNamespace: "cilium-secrets",
+		useProxyProtocol: true,
+	}
+	res := i.getHTTPRouteListener(&model.Model{
+		HTTP: []model.HTTPListener{
+			{
+				TLS: []model.TLSSecret{
+					{
+						Name:      "dummy-secret",
+						Namespace: "dummy-namespace",
+					},
+				},
+			},
+		},
+	})
+	require.Len(t, res, 1)
+	listener := &envoy_config_listener.Listener{}
+	err := proto.Unmarshal(res[0].GetValue(), listener)
+	require.NoError(t, err)
+
+	require.Len(t, listener.ListenerFilters, 2)
```

Comment:
I'm worried that this is a little fragile; it doesn't actually check that the extra filter is the correct one. Is it possible to add something to check that the second ListenerFilter is the ProxyProtocol one?